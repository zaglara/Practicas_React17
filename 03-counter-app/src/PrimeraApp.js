import React from 'react';
import PropTypes from 'prop-types';
//import React, { Fragment } from 'react';

const PrimeraApp = ({ saludo, subtitulo }) => {

    /*const saludo = {
        nombre: 'Leiza',
        apellido: 'Lara',
        edad: 24
    };*/

    /*const saludo = 'Hola Mundo';*/
    //console.log(saludo);

    return (
        <>
            <h1>{ saludo }</h1>
            {/*<pre>{ JSON.stringify(saludo, null, 3) }</pre>*/}
            <p>{ subtitulo }</p>
        </>
    );
    
}

PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired,
    subtitulo: PropTypes.string.isRequired
}

PrimeraApp.defaultProps = {
    subtitulo: 'Soy un subtitulo'
}

export default PrimeraApp;