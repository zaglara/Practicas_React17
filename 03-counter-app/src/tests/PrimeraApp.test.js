import React from 'react';
//import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import PrimeraApp from "../PrimeraApp";
import { shallow } from "enzyme";

describe('Pruebas en <Primera App />', () => { 
    /*test('debe de mostrar el mensaje "Hola, Soy Goku" ', () => { 
        const saludo = 'Hola, soy Goku';

        const { getByText } = render(<PrimeraApp saludo={saludo}/>);

        expect( getByText(saludo) ).toBeInTheDocument();
    });*/

    test('Debe mostrar <PrimeraApp /> correctamente', () => {
        const saludo = 'Hola, Soy Goku'; 
        const wrapper = shallow( <PrimeraApp saludo={saludo}/> );

        expect( wrapper ).toMatchSnapshot();
    });

    test('Debe mostrar el subtitulo enviado por props', () => { 
        const saludo = 'Hola, Soy Goku'; 
        const subTitulo = 'Soy el subtitulo de la prueba'; 
        const wrapper = shallow( 
            <PrimeraApp 
                saludo={ saludo }
                subtitulo={ subTitulo }
            /> 
        );
        const textoParrafo = wrapper.find('p').text();    //Equivale a querySelector
        console.log(textoParrafo);
        expect( textoParrafo ).toBe( subTitulo );
    });

 });