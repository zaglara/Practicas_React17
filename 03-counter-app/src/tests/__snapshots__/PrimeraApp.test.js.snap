// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Primera App /> Debe mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
    Hola, Soy Goku
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
